subsys/bluetooth/mesh/proxy_srv.c

```diff
@@ -90,34 +72,21 @@ static struct bt_mesh_proxy_client {
 		NONE,
 		WHITELIST,
 		BLACKLIST,
-		PROV,
 	} filter_type;
-#if defined(CONFIG_BT_MESH_GATT_PROXY)
 	struct k_work send_beacons;
-#endif /* CONFIG_BT_MESH_GATT_PROXY */
 } clients[CONFIG_BT_MAX_CONN] = {
 	[0 ... (CONFIG_BT_MAX_CONN - 1)] = {
-#if defined(CONFIG_BT_MESH_GATT_PROXY)
 		.send_beacons = Z_WORK_INITIALIZER(proxy_send_beacons),
-#endif /* CONFIG_BT_MESH_GATT_PROXY */
 		.cli.cb = {
 			.send = proxy_send,
-#if defined(CONFIG_BT_MESH_GATT_PROXY)
 			.recv = proxy_filter_recv,
```

Comment:
I think we discussed this in the previous PR as well, but IMO, it'd make sense to replace proxy_msg.c's `proxy_complete_pdu()` with this callback. This'd remove the last `IS_ENABLED()` macros from the common file, and wouldn't introduce any duplicate code at all.

I.e., instead of calling `proxy_complete_pdu()`, `bt_mesh_proxy_msg_recv()` would call this callback immediately, with the role and buffer pointers.

Reply:
Look like make sense to me 👍